torch/lib/c10d/Utils.cpp

```diff
@@ -131,6 +131,71 @@ std::pair<int, PortType> listen(PortType port) {
   return {socket, getSocketPort(socket)};
 }
 
+template <typename T>
+void handleConnectException(
+    struct ::addrinfo** nextAddr,
+    int error_code,
+    bool* anyRefused,
+    bool* anyReset,
+    bool wait,
+    T start,
+    std::shared_ptr<struct ::addrinfo> addresses,
+    std::chrono::milliseconds timeout) {
+  // ECONNREFUSED happens if the server is not yet listening.
+  if (error_code == ECONNREFUSED) {
+    *anyRefused = true;
+  }
+  // ECONNRESET happens if the server's listen backlog is exhausted.
+  if (error_code == ECONNRESET) {
```

Comment:
can `error_code` ever be something else? Should we throw if we get an error_code that is not one of these two?